### PR TITLE
doc : improved and cleanup of submitting-changes

### PR DESCRIPTION
**Pull request by mrVanDalo**
###### Motivation for this change

Some documentation refactoring

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

I don't know 

cc @

### CODE REVIEW

**Comment by mrVanDalo** [comment]
Don't get confused by my branch name.

**Comment by 7c6f434c** [comment]
Related to: nixos/rfcs#53

File: doc/submitting-changes.xml [hunk 1]
```diff
@@ -194,36 +187,17 @@ Additional information.
...
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     Follow <link xlink:href="https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes">the contribution guidelines</link>.
+    </para>
+   </listitem>
+   <listitem>
+    <para>
+     Notify maintainers of the package. For example add to the message: <command>cc @jagajaga @domenkozar</command>.
```
Comment (7c6f434c) [inline doc/submitting-changes.xml]: If you edit an existing package, ofBorg will notify the maintainers automatically, so maybe drop that?

Comment (mrVanDalo) [reply doc/submitting-changes.xml]: not yet. right @Lassulus  ?

Comment (Lassulus) [reply doc/submitting-changes.xml]: I think only the maintainer which are part of the organization. I'm not sure if this includes everyone by now?

Comment (mrVanDalo) [reply doc/submitting-changes.xml]: Also the cc part is still in the template. I looked at some pull-requests and it seems that ofBorg adds reviewers, but not all maintainers.

Comment (7c6f434c) [reply doc/submitting-changes.xml]: Asked in nixos/rfcs#39 — it looks like vast majority of maintainers are currently indeed organisation members (via readonly maintainers team) and can be review-requested. If every maintainer has already been invited, we could drop the cc line in template too, I guess.

Comment (mrVanDalo) [reply doc/submitting-changes.xml]: So I wouldn't do it now. I'll come back to it in another pull-request.

File: doc/submitting-changes.xml [hunk 2]
```diff
@@ -178,7 +171,7 @@ Additional information.
    </listitem>
    <listitem>
     <para>
-     Rebase you branch against current <command>master</command>.
+     <link xlink:href="https://git-scm.com/book/en/v2/Git-Branching-Rebasing">Rebase</link> you branch against current <command>master</command>.
```
Comment (alyssais) [inline doc/submitting-changes.xml]:
```suggestion
     <link xlink:href="https://git-scm.com/book/en/v2/Git-Branching-Rebasing">Rebase</link> your branch against current <command>master</command>.
```

**Review by alyssais — CHANGES_REQUESTED** [review]

**Comment by mrVanDalo** [comment]
@alyssais all done.

**Comment by Lassulus** [comment]
needs rebase

**Comment by mrVanDalo** [comment]
@Lassulus  done

File: doc/submitting-changes.xml (hunk 1 above)
Comment (7c6f434c) [reply doc/submitting-changes.xml]: I think NixOS/RFCs#39 has been now implemented as well as it can be…

Comment (mrVanDalo) [reply doc/submitting-changes.xml]: removed the line from this pull-request. I don't have the privileges to change the pull-request template. @Mic92 @Lassulus ?

**Comment by 7c6f434c** [comment]
>removed the line from this pull-request. I don't have the privileges to change the pull-request template. @Mic92 @Lassulus ?

Well, you can add a change to `<nixpkgs>/.github/PULL_REQUEST_TEMPLATE.md` to the PR

**Comment by mrVanDalo** [comment]
@7c6f434c done

> removed the line from this pull-request. I don't have the privileges to change the pull-request template. @Mic92 @Lassulus ?
> Well, you can add a change to `<nixpkgs>/.github/PULL_REQUEST_TEMPLATE.md` to the PR

**Review by 7c6f434c — APPROVED** [review]

**Pull request merged**